src/app/wizard/step/cluster/component.ts

```diff
@@ -480,6 +469,30 @@ export class ClusterStepComponent extends StepBase implements OnInit, ControlVal
     }
   }
 
+  private _updateAvailableProxyModes(): void {
+    const proxyModeControl = this.control(Controls.ProxyMode);
+    let newValue: ProxyMode;
+    if (this.controlValue(Controls.CNIPlugin) === CNIPlugin.Cilium) {
+      if (this.isKonnectivityEnabled && !!this.controlValue(Controls.Konnectivity)) {
+        this.availableProxyModes = [ProxyMode.iptables, ProxyMode.ebpf];
+        if (proxyModeControl.value === ProxyMode.ipvs) {
+          newValue = ProxyMode.ebpf;
+        }
+      } else {
+        this.availableProxyModes = [ProxyMode.iptables];
+        newValue = ProxyMode.iptables;
+      }
+    } else {
+      this.availableProxyModes = [ProxyMode.ipvs, ProxyMode.iptables];
+      if (proxyModeControl.pristine || !this.availableProxyModes.includes(proxyModeControl.value)) {
```

Comment:
Reason for adding `pristine` check is to avoid unintentional value changes when user switches between `Canal` and `Cilium`. When user switches to `Cilium`, `iptables` or `ebpf` proxy mode will be selected automatically but then if they switch back to `Canal` then `ipvs` will be reselected. Since Proxy Mode is hidden under Advanced settings so the user might not even notice this change (since they never manually changed Proxy Mode value).
This is demonstrated in attached video:


https://user-images.githubusercontent.com/13975988/180172378-1be2ea6e-f630-44e9-a1b9-26900099b353.mov